### ISSUE TITLE
[Calling] Callers should know if recipient's device is actually ringing

### ISSUE THREAD

**Issue by loxK** [signalapp/Signal-Android#11300]
### Bug description
When sending a video or audio call, the caller can't know if the recipient's device is actually ringing. We get the same ringing tone and app behavior that when the devices actually rings, in these cases:
- bug #7733
- recipient device is offline (discharged, out of coverage area etc.)

### Steps to reproduce
- shutdown recipient phone
- try to call it
- notice the same behavior that when recipient's device actually rings

**Comment by hiqua** [comment]
You can post feature requests on the forum: https://community.signalusers.org/c/feature-requests/android-feature-requests/18

Some argue for an "opposite" feature: https://community.signalusers.org/t/silent-call-reject/8990

**Comment by loxK** [comment]
It isn't an opposite feature at all. When calling, THE CALLER should know when it is actually ringing on the other side, muted or not. 

For example, I call someone: I get a ringing tone while connection is getting established, as soon as my contact's phone rings, I get to hear a different tone.

**Comment by hiqua** [comment]
I think you misunderstood my comment. At any rate, this discussion belongs to the forum.

**Comment by sivaraam** [comment]
There's actually a way for the caller to know when it is actually ringing, right now. The difference is not in the dial tone. The difference would be seen in the call status message shown below the contact's name in the dial screen. When it is trying to establish the call you would see "Calling". When it is actually ringing, you'll see "Ringing". If you use a different language, you should be seeing translations of those words.

In case your request is for a different dial tone for different states, kindly open a feature request in the community as correctly suggested by @hiqua

**Comment by sailbird** [comment]
By default, the caller _shouldn't_ be able to know whether or not the peer's device is making a sound -- only that it's aware of your attempt to connect (which is what, I assume, "Ringing" actually means). Otherwise an untrusted caller would be able to determine if the target phone is asleep, which might suggest that the owner is asleep or in a meeting. The caller has no business knowing that, even from a statistical perspective. At most, I would be in favor of an opt-in toggle that allows this information to be known.

**Comment by eabase** [comment]
> I would be in favor of an opt-in toggle that allows this information to be known

Obviously! A phone has one primary purpose, being able to place calls, and know that the person who you've tried to call will actually see that you called. Everything else is counterproductive paranoia. So YES, let's make that (anti-)feature as a setting. 

---

I can indeed confirm OP. 

Starting up phone P1 and calling it from P2. Will result in no ring signal at all! Signal is supposed to be (?) running in background. [You do hear "ring" signals in P2.] Opening signal on P1 and then closing it, and placing another call from P2, it will take about 3-4 "rings" before P2 is *actually* ringing, while you will hear ~6-7 in P2. 

Please prioritize this. :sparkling_heart: 

<sub> :heavy_check_mark:  I've been heavily endorsing Signal to everyone since the day this app first came out, so I expect certain basic things to work. :exclamation: </sub>

**Comment by loxK** [comment]
@sailbird @sivaraam as I said in a previous comment "The caller should know when it is actually ringing on the other side, **muted** or not.". I meant, like a normal phone call, a caller knows that its call is reaching the 
recipient, if the recipient's phone is in silent mode, DND mode or has muted the ring, belong to the recipient confidentiality and this is not what this request is about.

**Comment by Aeonitis** [comment]
Seems related to a concern raised 3 years ago: https://github.com/signalapp/Signal-Android/issues/8780

**Comment by Aeonitis** [comment]
If this is just about receiving a current incoming call and being notified, then #8780 is related.

**Comment by loxK** [comment]
bump

**Comment by htsmi** [comment]
I don't think this is a feature request, this is about the UI falsely saying the call is "ringing" when nothing is ringing on the other end. In may case I experienced #8780 which clearly needs to be addressed. Unfortunately I can't reproduce it right now to get a debug log. But according to others it also happens when the recipient's phone is off. I and others, I believe, have the expectation that when the UI says "Ringing" and the ringback tone plays that the app has received confirmation that the connection has been established and the other device is attempting to ring (either ringing or silently displaying a notification if it is silenced). This is how other apps and the traditional phone system work, so in this case I end up thinking my calls to friends and family are being ignored when they never rang in the first place.

**Comment by loxK** [comment]
The ringback tone plays even before showing "Ringing" on the screen. Without looking at the screen, it is impossible to know if the recipient's device is actually ringing (or silently displaying a notification if it is silenced).

**Comment by loxK** [comment]
bump

**Comment by breversa** [comment]
I believe the misunderstanding is about the word "ringing".

From what I understand (and agree with), there should be, on the **CALLER**’s side:
1. One tone while Signal is trying to establish connection between the caller and the callee
2. An OTHER tone (so that there’s no need to look at the screen) once the connection is established, and the callee’s phone is signaling (doesn’t matter if ringing, displaying a notification, both, something else, or nothing…) the incoming call.

Basically, the caller wants to know if the call is past the "establishing connection" stage.

**Comment by mr-brunes** [comment]
It may be useful to know that WhatsApp have finally fixed this behaviour i.e. when a call is at the 'calling' stage the caller hears no ringing tone - it is silent. The ringing tone is only heard by the caller when the call status goes to 'Ringing' i.e. the callee's phone is actually ringing.

The previous stance on this behaviour appeared to be to copy whatever Whatsapp did, so can we follow suit please?

**Comment by loxK** [comment]
The call screen now displays "ringing" when the recipient's device is actually ringing, so callers have better visibility into the call state (works on desktop too). This is definitely an improvement.

That said, adding a distinct ringing tone on the caller's side would be a valuable enhancement.

**Comment by mr-brunes** [comment]
>The call screen now displays "ringing" when the recipient's device is actually ringing,

The visual feedback has been like this for ages i.e. it's not a recent change. I'd guess that many (most) folks aren't monitoring the screen in any case for a slight wording change.
IMO the main issue is the misleading audio feedback i.e. the 'ringing' sound that the caller hears when nothing is ringing on the callee end.
